### PR TITLE
Migrate Export component to Vue3

### PR DESCRIPTION
**Pull request by tahmeedshah**
### General description
_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs
_Link to the related design prototypes (if applicable)_

### Testing instructions
_Provide minimal instructions on how to test this PR._

* Apart from the added feature / bug fix, check overall performance, styling...

## Checklist before merging
* [ ]  Branch name / PR includes the related Jira ticket Id.
* [ ]  Tests to check core implementation / bug fix added.
* [ ]  All checks in Continuous Integration workflow pass.
* [ ]  Feature functionally tested by reviewer(s).
* [ ]  Code reviewed by reviewer(s).
* [ ]  Documentation updated (README, CHANGELOG...) (if required)

### CODE REVIEW

File: app/components/common/export.vue [hunk 1]
```diff
@@ -54,24 +54,26 @@
...
                                                     <tr v-for="(row, index) in downloadDocs" :key="index">
                                                         
                                                         <td class="tableexport-string">{{row.rec_schema}}</td>
                                                         <td class="tableexport-string">
-                                                            <a rel="noopener" target="_blank" :href="`database/${$options.filters.encodeURIComponent($options.filters.capitalize((row.rec_uniqueIdentifier||'')))}`">
-                                                                {{(row.rec_uniqueIdentifier||'')|capitalize}}
+                                                            <!-- <a rel="noopener" target="_blank" :href="`database/${$options.filters.encodeURIComponent($options.filters.capitalize((row.rec_uniqueIdentifier||'')))}`"> -->
+                                                                <!-- ToDo: ?? -->
+                                                            <a v-if="row.rec_uniqueIdentifier" rel="noopener" target="_blank" :href="`database/${(row.rec_uniqueIdentifier).toUpperCase()}`">
```
Comment (sbilodeau) [inline app/components/common/export.vue:75]: I think it is missing `encodeURIComponent`
```suggestion
                                                            <a v-if="row.rec_uniqueIdentifier" rel="noopener" target="_blank" :href="`database/${encodeURIComponent((row.rec_uniqueIdentifier).toUpperCase())}`">
```

File: app/components/common/export.vue [hunk 2]
```diff
@@ -143,87 +149,86 @@ import '../kb/filters';
...
+                modal.show('static');
+                loading.value = true; 
                 try{
-                    const {docs, numFound, isGeneric, schemaFields,schema } = await this.getDownloadRecords({fields, listType:'initial', format:'json'});
-                    this.downloadDocs = docs
-                    this.numFound     = numFound;
-                    this.isGeneric    = isGeneric;
-                    this.schemaFields = schemaFields;
-                    this.schema       = schema;                   
+                        console.log('getDownloadRecords', typeof getDownloadRecords) // type of getDownloadRecords object, it should be function.
```
Comment (sbilodeau) [inline app/components/common/export.vue:176]: can drop this line
```suggestion
```

File: app/components/common/export.vue [hunk 3]
```diff
@@ -143,87 +149,86 @@ import '../kb/filters';
...
-		async mounted() {
-            this.modal = new Modal(this.$refs.exportModal);
-        },
-		
-		methods: {
-            async openDialog(){
-                this.modal.show('static');
-                this.loading      = true; 
+    ];
+    const exportModal = ref(null);
```
Comment (sbilodeau) [inline app/components/common/export.vue:153]: i think for dom `refs` (from line 7) it's better to use `shallowRef` as the reactivity will be only on the reference and not on the dom elelment OR component

https://vuejs.org/api/reactivity-advanced.html#reactivity-api-advanced
```suggestion
    const exportModal = shallowRef(null);
```

File: app/views/search/directives/result-view-options.js [hunk 4]
```diff
@@ -23,8 +24,37 @@ app.directive('resultViewOptions', ['$location', 'ngDialog', 'locale', 'apiToken
...
+                }
+
+                function safeApply ($scope, fn) {
+                    const phase = $scope.$root.$$phase;
+                  
+                    if (phase === '$apply' || phase === '$digest') {
+                      return fn();
+                    } else {
+                      return $scope.$apply(fn);
+                    }
```
Comment (sbilodeau) [inline app/views/search/directives/result-view-options.js:57]: can you import from `angular-vue` or extract to the services folder?

**Review by sbilodeau — CHANGES_REQUESTED** [review]

File: app/components/common/export.vue (hunk 1 above)
Comment (tahmeedshah) [reply app/components/common/export.vue]: added encodeURIComponent

File: app/views/search/directives/result-view-options.js (hunk 4 above)
Comment (tahmeedshah) [reply app/views/search/directives/result-view-options.js]: moved safeApply to services/common.js

**Review by blaisefonseca — APPROVED** [review]

**Review by sbilodeau — APPROVED** [review]

**Pull request merged**